src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -39,6 +39,28 @@ public static bool TryGetArray<T>(ReadOnlyMemory<T> readOnlyMemory, out ArraySeg
             return false;
         }
 
+        /// <summary>
+        /// Get a <see cref="OwnedMemory{T}"/> from the underlying memory.
+        /// If unable to get the <typeparamref name="TOwner"/>, return false with a default <typeparamref name="TOwner"/>.
+        /// </summary>
+        public static bool TryGetOwnedMemory<T, TOwner>(ReadOnlyMemory<T> readOnlyMemory, out TOwner ownedMemory)
+            where TOwner : OwnedMemory<T>
+        {
+            ownedMemory = readOnlyMemory.GetObjectStartLength(out int index, out int length) as TOwner;
+            return !ReferenceEquals(ownedMemory, null);
```

Comment:
Does the JIT'd code end up fetching ownedMemory again?  Wondering if there's any value in this being:
```C#
TOwner tmp;
ownedMemory = tmp = readOnlyMemory.GetObjectStartLength(out int index, out int length) as TOwner;
return !ReferenceEquals(tmp, null);
```

Reply:
Ah, so its reg rather than ref; good idea